### PR TITLE
Adding the fetch SVG API and adding fixes and improvements (v1)

### PR DESCRIPTION
**Pull request by iifawzi**
### Description

Finally we're here, the first API should be ready to work after the migration! 

Fetching SVG icons (v1) should work like a charm now, fixed the most of the bugs I've noticed and handled the cases that crossed my mind, it can be used to fetch single, multiple, with or without customizations - SVG icons. 
No changes is made, it will function as it was suppose to function.

### CODE REVIEW

File: src/utils/tools/getThemeDir.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    themeDir = ThemesDirectories[iconsTheme.filled]
+  } else if (theme === 'svg-outlined') {
+    themeDir = ThemesDirectories[iconsTheme.outlined]
+  } else {
+    themeDir = ThemesDirectories[theme]
+  }
+  return themeDir
+}
+
+export default getThemeDir
```
Comment (iifawzi) [inline src/utils/tools/getThemeDir.ts:22]: I've moved the process of determining the correct directory to a separate tool, since it will be used in multiple places across the code.

File: src/utils/svgFactory.ts [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
 
-    async finalizeIcon () {
+    finalizeIcon () {
       this.readSvgCode()
-      this.modifySvgCode()
-      await pfs.writeFile(this.outputPathToSvg, this.svgCode)
+      if (this.customized) {
+        this.modifySvgCode()
+      }
+      return this.svgCode
```
Comment (iifawzi) [inline src/utils/svgFactory.ts:44]: This's added to handle the case if no customization is required, an undefined error was being thrown before.

File: src/utils/svgFactory.ts [hunk 3]
```diff
@@ -1,45 +1,47 @@
...
       const index1 = this.svgCode.indexOf('>') + 1
       const index2 = this.svgCode.lastIndexOf('<')
       this.svgCode = `${this.svgCode.slice(0, index1)}${gTagOpen}${this.svgCode.slice(index1, index2)}</g></svg>`
     }
 
-    async finalizeIcon () {
+    finalizeIcon () {
       this.readSvgCode()
-      this.modifySvgCode()
-      await pfs.writeFile(this.outputPathToSvg, this.svgCode)
```
Comment (iifawzi) [inline src/utils/svgFactory.ts:42]: Writing process will be moved to the controllers, and I don't think so it will even be used, for the `svgToPng` it seems like it can take the `svg` code and convert it directly, I will make the required changes to the `ImageFactory` class

Comment (iifawzi) [reply src/utils/svgFactory.ts:42]: ok it seems like, the `icons-customization` API, can be used to download an `svg` icons with customization also, this's can be considered as a duplication, but since we're not going to update the old routes, I will keep it as it's. 

After the full deprecation of `v1` API, we can then update this Factory.

File: src/utils/tools/getThemeDir.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// *********************************
+// This function is used to return the appropriate directory, based on the theme:
+// we're handling the case of `svg` and `svg-outlined` manually for now, until the full
+// depreciation of version 1 APIs
+// *********************************
+
+const getThemeDir = (theme: iconsTheme | iconsThemeV1 = iconsTheme.filled) => {
+  let themeDir: string
+  if (theme === 'svg') {
+    themeDir = ThemesDirectories[iconsTheme.filled]
```
Comment (kaybhutani) [inline src/utils/tools/getThemeDir.ts:13]: one question. why is this needed in the first place? if we provide the enums filled, outlined values of `svg & svg-outlined` resp, we can use it directly without these conditions right?

File: src/utils/svgFactory.ts [hunk 5]
```diff
@@ -1,45 +1,47 @@
...
       if (fs.existsSync(this.pathToSvg)) {
         this.svgCode = fs.readFileSync(this.pathToSvg, 'utf8')
       } else {
         this.svgCode = fs.readFileSync(this.pathToSvg.replace('svg-outlined', 'svg'), 'utf8')
       }
     }
 
-    modifySvgCode () {
+    private modifySvgCode () {
       const gTagOpen = `<g ${this.changeColor()} transform="${this.rotateIcon()} ${this.flipIcon()}">`
```
Comment (kaybhutani) [inline src/utils/svgFactory.ts]: So, here is an edge case (we can fix this later), in certain scenarios when rotate and flip both are applied, this customization wont work as we dont know what was applied first. (happens usually with asymmetric icons) but we should try to fix this later. Let's talk about this over chat

File: src/utils/tools/getThemeDir.ts (hunk 4 above)
Comment (iifawzi) [reply src/utils/tools/getThemeDir.ts:13]: This's just a backward compatibility, `v1` apis are using `svg` and `svg-outlined`, as per our conversation #12, we're using `filled` and `outlined`, that's way we're just mapping the old values to use the directories we specified in `filled` and `outlined` until the full deprecation of `v1`.

Have I answered this question? i'm not sure if I understood your question correctly.

File: src/utils/svgFactory.ts (hunk 5 above)
Comment (iifawzi) [reply src/utils/svgFactory.ts]: Hmm, ok, graphically, I think flip should be applied first in all cases, then we can make the rotation, I'm not really sure.

File: src/utils/svgFactory.ts (hunk 3 above)
Comment (iifawzi) [reply src/utils/svgFactory.ts:42]: I'm working on a refactored version of `imageFactory` class, that would handle all cases efficiently, in fact, we don't need to create an svg files, then deleting them. `svgToPng` can take the `svg` string directly and convert it to `png`
I wil make the changes on #14
